app/controllers/concerns/two_factor_authenticatable_methods.rb

```diff
@@ -47,6 +47,7 @@ def check_already_authenticated
     return unless initial_authentication_context?
     return unless user_fully_authenticated?
     return if remember_device_expired_for_sp?
+    return if (aal3_policy.aal3_required? && mfa_policy.aal3_mfa_enabled? && !aal3_policy.aal3_used?)
```

Comment:
can we make this into a method? So that we could name it to clarify what it does?

Maybe
```ruby
return if aa3_required_but_not_used?
```